Workaround for excessively long gcc command line build issue

dmitri-d opened the pull request:
This is a port of scripts and patches used during proxy rpm package build

dmitri-d commented:
> Why this can't be checked-in in tree? So that an user building it locally could benefit?

Upstream didn't update include paths until a later envoy version. Should we modify paths in the codebase directly, none of upstream changes (bug fixes and CVE fixes) would apply without changes.

> You did it on envoy side already, no?

Not yet.

jwendell commented on maistra/ci/post-submit.sh:
```diff
@@ -12,6 +12,9 @@ DIR=$(cd "$(dirname "$0")" ; pwd -P)
 source "${DIR}/common.sh"
 
+# modify include paths to avoid overly long gcc command line
+fix_include_paths()
```
you can move this call (without `()`) to the common.sh file

dmitri-d replied:
This and below: not sure what you mean -- paths should only be modified once per checkout, can't invoke the update from bazel_build/bazel_test...

jwendell replied:
You are calling this function in both files. You can just move this call to the bottom of common.sh, which is included in pre and post submit files.

dmitri-d replied:
done

jwendell commented on maistra/ci/pre-submit.sh:
```diff
@@ -12,6 +12,9 @@ DIR=$(cd "$(dirname "$0")" ; pwd -P)
 source "${DIR}/common.sh"
 
+# modify include paths to avoid overly long gcc command line
+fix_include_paths
```
you can move this call to the common.sh file

jwendell reviewed (commented):
ah ok.
so in https://github.com/maistra/envoy/pull/98 we added the envoy files but are not using them? should it be reverted in favor of this one then?

dmitri-d commented:
> so in maistra/envoy#98 we added the envoy files but are not using them? should it be reverted in favor of this one then?

Lol, I completely forgot about it, updated the PR to use the patch and the script in maistra-envoy repository.

dmitri-d commented:
@jwendell: the build is going to fail, there's a syntax error in the script used to update paths in envoy (the fix is here: https://github.com/maistra/envoy/pull/106)

dmitri-d commented:
/retest

jwendell commented:
/retest
